### PR TITLE
refactor(test): handling assert fail messages in tests

### PR DESCRIPTION
**Pull request by AnthonyGlt**
## Description
Catch error when assert is false and avoid timeout.

## Motivation and Context
I was getting a timeout when the assert was wrong for unit tests.

Example of timeout: demutils.js

```
it('add elevation layer', (done) => {
        viewer.addLayer(elevationlayer)
            .then((l) => {
                assert.equal('worldelevation1', l.id);
                done();
            }, done);
    });
 

``` 
I've replaced _worldelevation_ with _worldelevation1_ and it will timeout instead of returning the error.

Following the commit from:
https://github.com/iTowns/itowns/pull/2044

### CODE REVIEW

File: test/unit/featuregeometrylayererror.js [hunk 1]
```diff
@@ -72,8 +72,8 @@ files.forEach((geojson, i) => {
                         .then(() => {
                             assert.equal(layerProj4.object3d.children.length, 1);
                             done();
-                        }, done);
-                }, done);
+                        }).catch(done);
```
Comment (ftoromanoff) [inline test/unit/featuregeometrylayererror.js]: Is this catch() really neccessary as there is another catch() just after ?

File: test/unit/potree.js [hunk 2]
```diff
@@ -99,8 +99,8 @@ describe('Potree', function () {
                         .then(() => {
                             assert.equal(2, root.children[0].children.length);
                             done();
-                        });
-                }, done);
+                        }).catch(done);
+                }).catch(done);
```
Comment (ftoromanoff) [inline test/unit/potree.js]: same

File: test/unit/featuregeometrylayererror.js (hunk 1 above)
Comment (AnthonyGlt) [reply test/unit/featuregeometrylayererror.js]: We need to catch each _then_ error or else the parents' _then_ will timeout. Like this, the error is catched by each _then_ and returned.

Comment (AnthonyGlt) [reply test/unit/featuregeometrylayererror.js]: I will do a _return_ when I can

**Review by ftoromanoff — APPROVED** [review]

**Pull request merged**